tests/unit/test_parallel_write.py

```diff
@@ -0,0 +1,271 @@
+    with (
+        patch("sys.stderr", new=StringIO()) as tqdm_out,
+        ZarrIO(path=zarr_top_level_path,  manager=get_manager(), mode="w") as io
+    ):
+        pickleable_column = VectorData(
+            name="TestPickleableIteratorColumn",
+            description="",
+            data=PickleableDataChunkIterator(
+                data=np.array([1., 2., 3.]),
+                display_progress=True,
+            )
+        )
+        not_pickleable_column = VectorData(
+            name="TestNotPickleableColumn",
+            description="",
+            data=NotPickleableDataChunkIterator(
+                data=np.array([4., 5., 6.]),
+                display_progress=True,
+                progress_bar_options=dict(desc=expected_desc_not_pickleable, position=1)
+            )
+        )
+        dynamic_table = DynamicTable(
+            name="TestTable", description="", columns=[pickleable_column, not_pickleable_column]
+        )
+        io.write(container=dynamic_table, number_of_jobs=number_of_jobs)
+
+    tqdm_out_value = tqdm_out.getvalue()
+    assert expected_desc_pickleable in tqdm_out_value
+    assert expected_desc_not_pickleable in tqdm_out_value
+
+
+def test_extra_keyword_argument_propagation(tmpdir):
+    number_of_jobs = 2
+
+    column = VectorData(name="TestColumn", description="", data=np.array([1., 2., 3.]))
+    dynamic_table = DynamicTable(name="TestTable", description="", id=list(range(3)), columns=[column])
+
+    zarr_top_level_path = str(tmpdir / "test_extra_parallel_write_keyword_arguments.zarr")
+
+    test_keyword_argument_pairs = [
+        dict(max_threads_per_process=2, multiprocessing_context=None),
+        dict(max_threads_per_process=None, multiprocessing_context="spawn"),
+        dict(max_threads_per_process=2, multiprocessing_context="spawn"),
+    ]
+    if platform.system() != "Windows":
+        test_keyword_argument_pairs.extend(
+            [
+                dict(max_threads_per_process=None, multiprocessing_context="spawn"),
+                dict(max_threads_per_process=2, multiprocessing_context="spawn"),
+            ]
+        )
+
+    for test_keyword_argument_pair in test_keyword_argument_pairs:
+        test_max_threads_per_process = test_keyword_argument_pair["max_threads_per_process"]
+        test_multiprocessing_context = test_keyword_argument_pair["multiprocessing_context"]
+        with ZarrIO(path=zarr_top_level_path,  manager=get_manager(), mode="w") as io:
+            io.write(
+                container=dynamic_table,
+                number_of_jobs=number_of_jobs,
+                max_threads_per_process=test_max_threads_per_process,
+                multiprocessing_context=test_multiprocessing_context
+            )
+
+            assert io._ZarrIO__dci_queue.max_threads_per_process == test_max_threads_per_process
+            assert io._ZarrIO__dci_queue.multiprocessing_context == test_multiprocessing_context
```

Comment:
Out of curiosity, can anyone direct me to an explanation of _why_ the `ZarrIO.__dci_queue` attribute ends up as `io._ZarrIO__dci_queue` here?

Reply:
That is just Python's way of supporting `private` member variables and functions by mangling names. This part of the Python docs explains this briefly https://docs.python.org/3/tutorial/classes.html#private-variables

Reply:
Thanks for the pointer

Learn something new every day lol